src/PowerPing/Ping.cs

```diff
@@ -415,38 +399,54 @@ private PingResults SendICMP(PingAttributes attrs)
+                        }
                         replyTime = new TimeSpan(Helper.StopwatchToTimeSpanTicks(Stopwatch.GetTimestamp() - requestTimestamp));
 
-                        // Store reply packet
-                        response = new ICMP(receiveBuffer, bytesRead);
-
-                        // Ignore unexpected echo responses
-                        if (packet.type == 8 && response.type == 0) {
-                            ushort responseSessionId = BitConverter.ToUInt16(response.message, 0);
-                            ushort responseSequenceNum = BitConverter.ToUInt16(response.message, 2);
-                            if (responseSessionId != sessionId || responseSequenceNum != sequenceNum) {
-                                if (replyTime.TotalMilliseconds >= attrs.Timeout) {
-                                    throw new SocketException();
+                        if (bytesRead == 0) {
+                            response = null;
+                        } else {
+                            // Store reply packet
+                            response = new ICMP(receiveBuffer, bytesRead);
+
+                            // Ignore unexpected echo responses
+                            if (packet.type == 8 && response.type == 0) {
+                                ushort responseSessionId = BitConverter.ToUInt16(response.message, 0);
+                                ushort responseSequenceNum = BitConverter.ToUInt16(response.message, 2);
+                                if (responseSessionId != sessionId || responseSequenceNum != sequenceNum) {
+                                    response = null;
                                 }
-                                response = null;
                             }
                         }
                     } while (response == null);
 
                     // Display reply packet
                     if (Display.ShowReplies) {
-                        PowerPing.Display.ReplyPacket(response, Display.UseInputtedAddress | Display.UseResolvedAddress ? attrs.Host : ep.ToString(), index, replyTime, bytesRead);
+                        PowerPing.Display.ReplyPacket(response, Display.UseInputtedAddress | Display.UseResolvedAddress ? attrs.Host : responseEP.ToString(), index, replyTime, bytesRead);
                     }
 
                     // Store response info
-                    try { Results.Received++; }
-                    catch (OverflowException) { Results.HasOverflowed = true; }
-                    Results.CountPacketType(response.type);
-                    Results.SaveResponseTime(replyTime.TotalMilliseconds);
+                    try { results.Received++; }
+                    catch (OverflowException) { results.HasOverflowed = true; }
+                    results.CountPacketType(response.type);
+                    results.SaveResponseTime(replyTime.TotalMilliseconds);
                     
 		            if (attrs.BeepLevel == 2) {
```

Comment:
Indentations a bit weird here. Don't know how long thats been that way for 😛

Reply:
Yeah it was already like that, you have some tabs mixed in with your spaces.